addons/point_of_sale/static/src/js/models.js

```diff
@@ -1465,12 +1465,18 @@ class PosGlobalState extends PosModel {
      */
     async _addProducts(ids, setAvailable=true){
         if(setAvailable){
-            await this.env.services.rpc({
-                model: 'product.product',
-                method: 'write',
-                args: [ids, {'available_in_pos': true}],
-                context: this.env.session.user_context,
-            });
+            try {
+                await this.env.services.rpc({
+                    model: 'product.product',
+                    method: 'write',
+                    args: [ids, {'available_in_pos': true}],
+                    context: this.env.session.user_context,
+                });
+            } catch (error) {
+                if (!this._isRPCError(error) || (error.message.data && error.message.data.name !== 'odoo.exceptions.AccessError')) {
+                    throw error;
+                }
```

Comment:
```suggestion
                if (this._isRPCError(error) && error.message.data && error.message.data.name === 'odoo.exceptions.AccessError') {
                    return;
                }
                throw error;
```
I think this is more readable.